# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/validationGL43_autogen.h
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/libANGLE/validationGL43_autogen.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/gfx/angle/checkout/src/libANGLE/validationGL43_autogen.h	Fri Nov 08 17:06:00 2019 +0000
@@ -0,0 +1,167 @@
+// GENERATED FILE - DO NOT EDIT.
+// Generated by generate_entry_points.py using data from gl.xml and wgl.xml.
+//
+// Copyright 2019 The ANGLE Project Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+//
+// validationGL43_autogen.h:
+//   Validation functions for the OpenGL 4.3 entry points.
+
+#ifndef LIBANGLE_VALIDATION_GL43_AUTOGEN_H_
+#define LIBANGLE_VALIDATION_GL43_AUTOGEN_H_
+
+#include "common/PackedEnums.h"
+
+namespace gl
+{
+class Context;
+
+bool ValidateClearBufferData(Context *context,
+                             GLenum target,
+                             GLenum internalformat,
+                             GLenum format,
+                             GLenum type,
+                             const void *data);
+bool ValidateClearBufferSubData(Context *context,
+                                GLenum target,
+                                GLenum internalformat,
+                                GLintptr offset,
+                                GLsizeiptr size,
+                                GLenum format,
+                                GLenum type,
+                                const void *data);
+bool ValidateCopyImageSubData(Context *context,
+                              GLuint srcName,
+                              GLenum srcTarget,
+                              GLint srcLevel,
+                              GLint srcX,
+                              GLint srcY,
+                              GLint srcZ,
+                              GLuint dstName,
+                              GLenum dstTarget,
+                              GLint dstLevel,
+                              GLint dstX,
+                              GLint dstY,
+                              GLint dstZ,
+                              GLsizei srcWidth,
+                              GLsizei srcHeight,
+                              GLsizei srcDepth);
+bool ValidateDebugMessageCallback(Context *context, GLDEBUGPROC callback, const void *userParam);
+bool ValidateDebugMessageControl(Context *context,
+                                 GLenum source,
+                                 GLenum type,
+                                 GLenum severity,
+                                 GLsizei count,
+                                 const GLuint *ids,
+                                 GLboolean enabled);
+bool ValidateDebugMessageInsert(Context *context,
+                                GLenum source,
+                                GLenum type,
+                                GLuint id,
+                                GLenum severity,
+                                GLsizei length,
+                                const GLchar *buf);
+bool ValidateGetDebugMessageLog(Context *context,
+                                GLuint count,
+                                GLsizei bufSize,
+                                GLenum *sources,
+                                GLenum *types,
+                                GLuint *ids,
+                                GLenum *severities,
+                                GLsizei *lengths,
+                                GLchar *messageLog);
+bool ValidateGetInternalformati64v(Context *context,
+                                   GLenum target,
+                                   GLenum internalformat,
+                                   GLenum pname,
+                                   GLsizei bufSize,
+                                   GLint64 *params);
+bool ValidateGetObjectLabel(Context *context,
+                            GLenum identifier,
+                            GLuint name,
+                            GLsizei bufSize,
+                            GLsizei *length,
+                            GLchar *label);
+bool ValidateGetObjectPtrLabel(Context *context,
+                               const void *ptr,
+                               GLsizei bufSize,
+                               GLsizei *length,
+                               GLchar *label);
+bool ValidateGetProgramResourceLocationIndex(Context *context,
+                                             GLuint program,
+                                             GLenum programInterface,
+                                             const GLchar *name);
+bool ValidateInvalidateBufferData(Context *context, GLuint buffer);
+bool ValidateInvalidateBufferSubData(Context *context,
+                                     GLuint buffer,
+                                     GLintptr offset,
+                                     GLsizeiptr length);
+bool ValidateInvalidateTexImage(Context *context, GLuint texture, GLint level);
+bool ValidateInvalidateTexSubImage(Context *context,
+                                   GLuint texture,
+                                   GLint level,
+                                   GLint xoffset,
+                                   GLint yoffset,
+                                   GLint zoffset,
+                                   GLsizei width,
+                                   GLsizei height,
+                                   GLsizei depth);
+bool ValidateMultiDrawArraysIndirect(Context *context,
+                                     GLenum mode,
+                                     const void *indirect,
+                                     GLsizei drawcount,
+                                     GLsizei stride);
+bool ValidateMultiDrawElementsIndirect(Context *context,
+                                       GLenum mode,
+                                       GLenum type,
+                                       const void *indirect,
+                                       GLsizei drawcount,
+                                       GLsizei stride);
+bool ValidateObjectLabel(Context *context,
+                         GLenum identifier,
+                         GLuint name,
+                         GLsizei length,
+                         const GLchar *label);
+bool ValidateObjectPtrLabel(Context *context, const void *ptr, GLsizei length, const GLchar *label);
+bool ValidatePopDebugGroup(Context *context);
+bool ValidatePushDebugGroup(Context *context,
+                            GLenum source,
+                            GLuint id,
+                            GLsizei length,
+                            const GLchar *message);
+bool ValidateShaderStorageBlockBinding(Context *context,
+                                       GLuint program,
+                                       GLuint storageBlockIndex,
+                                       GLuint storageBlockBinding);
+bool ValidateTexBufferRange(Context *context,
+                            GLenum target,
+                            GLenum internalformat,
+                            GLuint buffer,
+                            GLintptr offset,
+                            GLsizeiptr size);
+bool ValidateTexStorage3DMultisample(Context *context,
+                                     TextureType targetPacked,
+                                     GLsizei samples,
+                                     GLenum internalformat,
+                                     GLsizei width,
+                                     GLsizei height,
+                                     GLsizei depth,
+                                     GLboolean fixedsamplelocations);
+bool ValidateTextureView(Context *context,
+                         GLuint texture,
+                         GLenum target,
+                         GLuint origtexture,
+                         GLenum internalformat,
+                         GLuint minlevel,
+                         GLuint numlevels,
+                         GLuint minlayer,
+                         GLuint numlayers);
+bool ValidateVertexAttribLFormat(Context *context,
+                                 GLuint attribindex,
+                                 GLint size,
+                                 GLenum type,
+                                 GLuint relativeoffset);
+}  // namespace gl
+
+#endif  // LIBANGLE_VALIDATION_GL43_AUTOGEN_H_